src/inetstack/mod.rs

```diff
@@ -268,10 +270,12 @@ impl NetworkTransport for SharedInetStack {
     async fn push(
         &mut self,
         sd: &mut Self::SocketDescriptor,
-        buf: &mut DemiBuffer,
+        bufs: ArrayVec<DemiBuffer, DEMI_SGARRAY_MAXLEN>,
         addr: Option<SocketAddr>,
     ) -> Result<(), Fail> {
-        self.layer4_endpoint.push(sd, buf, addr).await
+        debug_assert!(DEMI_SGARRAY_MAXLEN >= MAX_BATCH_SIZE_NUM_PACKETS);
```

Comment:
Can this be done outside the critical path (even though it's only compiled in for debug mode)? Both operands are consts, so this should be checked just once, if possible.

Reply:
yes, let me check if there is a compile assert that I can use

Reply:
So it looks like there is a static assert crate but we might not want to add that to the repo, so then the next best option is this: https://internals.rust-lang.org/t/nicer-static-assertions/15986/2